envi/archs/aarch64/const.py

```diff
@@ -0,0 +1,1730 @@
+    0b10010010110101: 'RAVLE2NXS',
+    0b10010010111000: 'ALLE2NXS',
+    0b10010010111001: 'VAE2NXS',
+    0b10010010111100: 'ALLE1NXS',
+    0b10010010111101: 'VALE2NXS',
+    0b10010010111110: 'VMALLS12E1NXS',
+    0b11010010001000: 'ALLE3OSNXS',
+    0b11010010001001: 'VAE3OSNXS',
+    0b11010010001101: 'VALE3OSNXS',
+    0b11010010010001: 'RVAE3ISNXS',
+    0b11010010010101: 'RVALE3ISNXS',
+    0b11010010011000: 'ALLE3ISNXS',
+    0b11010010011001: 'VAE3ISNXS',
+    0b11010010011101: 'VALE3ISNXS',
+    0b11010010101001: 'RVAE3OSNXS',
+    0b11010010101101: 'RVALE3OSNXS',
+    0b11010010110001: 'RVAE3NXS',
+    0b11010010110101: 'RVALE3NXS',
+    0b11010010111000: 'ALLE3NXS',
+    0b11010010111001: 'VAE3NXS',
+    0b11010010111101: 'VALE3NXS',
+    # Following require FEAT_RME:
+    0b11010000001100: 'PAALLOS',
+    0b11010000100011: 'RPAOS',
+    0b11010000100111: 'RPALOS',
+    0b11010000111100: 'PAALL'
+}
+
+# Barrier operation assembler symbols, encoded in crm
+bar_ops_table = {
+    0b1111: 'sy',
+    0b1110: 'st',
+    0b1101: 'ld',
+    0b1011: 'ish',
+    0b1010: 'ishst',
+    0b1001: 'ishld',
+    0b0111: 'nsh',
+    0b0110: 'nshst',
+    0b0101: 'nshld',
+    0b0011: 'osh',
+    0b0010: 'oshst',
+    0b0001: 'oshld'
+}
+
+psb_op = {
+    0b0: 'csync'
+}
+
+
+sys_alias_op_tables = (
+    at_op_table,
+    dc_op_table,
+    ic_op_table,
+    tlbi_op_table,
+    bar_ops_table,
+    psb_op, 
+)
+
+barrier_option_table = (
+    0,
+    'OSHLD',
+    'OSHST',
+    'OSH',
+    4,
+    'NSHLD',
+    'NSHST',
+    'NSH',
+    8,
+    'ISHLD',
+    'ISHST',
+    'ISH',
+    12,
+    'LD',
+    'ST',
+    'SY',
+)
+
+b_cond_table = (
+    ('b.eq', INS_BCC),
+    ('b.ne', INS_BCC),
+    ('b.hs', INS_BCC),
+    ('b.lo', INS_BCC),
+    ('b.mi', INS_BCC),
+    ('b.pl', INS_BCC),
+    ('b.vs', INS_BCC),
+    ('b.vc', INS_BCC),
+    ('b.hi', INS_BCC),
+    ('b.ls', INS_BCC),
+    ('b.ge', INS_BCC),
+    ('b.lt', INS_BCC),
+    ('b.gt', INS_BCC),
+    ('b.le', INS_BCC),
+    ('b.al', INS_BCC),
+    ('b.nv^B', INS_BCC),
+)
+
+cond_table = (
+    'EQ',
+    'NE',
+    'CS',
+    'CC',
+    'MI',
+    'PL',
+    'VS',
+    'VC',
+    'HI',
+    'LS',
+    'GE',
+    'LT',
+    'GT',
+    'LE',
+    'AL',
+    'NV^b', #FIXME
+)
+
+COND_EQ = 0
+COND_NE = 1
+COND_CS = 2
+COND_CC = 3
+COND_MI = 4
+COND_PL = 5
+COND_VS = 6
+COND_VC = 7
+COND_HI = 8
+COND_LS = 9
+COND_GE = 10
+COND_LT = 11
+COND_GT = 12
+COND_LE = 13
+COND_AL = 14
+COND_NV = 15
+
+inv_cond_table = (
+        COND_NE,
+        COND_EQ,
+        COND_CC,
+        COND_CS,
+        COND_PL,
+        COND_MI,
+        COND_VC,
+        COND_VS,
+        COND_LS,
+        COND_HI,
+        COND_LT,
+        COND_GE,
+        COND_LE,
+        COND_GT,
+        COND_NV,
+        COND_AL,
+)
+
+
+prefetchTypes = ["pld", "pli", "pst"]   
+prefetchTargets = ["l1", "l2", "l3"]
+prefetchPolicy = ["keep", "strm"]
+
+# Pstatefield names for MSR using the "op1:op2:CRm" encoding
+pstatefield_names = {
```

Comment:
This seems like a lot of repetition to introduce a couple mappings and feels like we simplify this down?